### PR TITLE
docs: fix Markdown link

### PR DESCRIPTION
**Pull request by jbampton**
### What this PR does / why we need it:

Standardizes the Markdown files.

### Pre-submission checklist:

* [X] Did you explain what problem does this PR solve? Or what new features have been added?
* [ ] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [ ] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: doc/zh-cn/discovery.md [hunk 1]
```diff
@@ -23,12 +23,12 @@
 * [**摘要**](#摘要)
 * [**如何扩展注册中心**](#如何扩展注册中心)
     * [**基本步骤**](#基本步骤)
-    * [**以 Eureka 举例**](#以-Eureka-举例)
+    * [**以 Eureka 举例**](#以-eureka-举例)
         * [**实现 eureka.lua**](#实现-eurekalua)
-        * [**Eureka 与 APISIX 之间数据转换逻辑**](#Eureka-与-APISIX-之间数据转换逻辑)
+        * [**Eureka 与 APISIX 之间数据转换逻辑**](#eureka-与-APISIX-之间数据转换逻辑)
```
Comment (juzhiyuan) [inline doc/zh-cn/discovery.md:28]: Why we need lowercase?

Comment (jbampton) [reply doc/zh-cn/discovery.md:28]: The old syntax does work but most of the codebase is using lowercase anchors.  

We should have one standard and consistent way.

Comment (juzhiyuan) [reply doc/zh-cn/discovery.md:28]: good good

**Review by juzhiyuan — APPROVED** [review]

File: doc/zh-cn/discovery.md [hunk 2]
```diff
@@ -23,12 +23,12 @@
 * [**摘要**](#摘要)
 * [**如何扩展注册中心**](#如何扩展注册中心)
     * [**基本步骤**](#基本步骤)
-    * [**以 Eureka 举例**](#以-Eureka-举例)
+    * [**以 Eureka 举例**](#以-eureka-举例)
```
Comment (spacewander) [inline doc/zh-cn/discovery.md:26]: I think it is fine to use `Eureka`? As we use Python instead of python.

**Review by spacewander — CHANGES_REQUESTED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**